### PR TITLE
Implement pg_autoctl show ipaddr|cidr.

### PR DESCRIPTION
**Pull request by DimCitus**
Rather than using `hostname -fqdn` we introduce new commands `show ipaddr`
which displays the discovered LAN ip address, and `show cidr` which allows
for debugging pg_autoctl HBA edits.

Those commands would be nice additions to the PG_AUTOCTL_DEBUG mode for the
command line, but it seems that in some environments with broken reverse DNS
we need an easy way to discover the LAN IP address from the command line. A
quick search shows it's pretty hard to do that with classic OS tooling in a
portable way, and we already have had to write code for that feature in
pg_auto_failover anyway. This patch exposes the feature to the end users.

See #6.

### CODE REVIEW

File: docs/ref/reference.rst [hunk 1]
```diff
@@ -207,6 +209,21 @@ commands can be used, from any node in the setup.
     For details about the options to the command, see above in the ``pg_autoctl
     show events`` command.
 
+  - ``pg_autoctl show uri``
```
Comment (Bogdanp) [inline docs/ref/reference.rst]: Please excuse the drive-by comment, but it looks like this might have been mistakenly copied from above. This should say `pg_autoctl show ipaddr`, no?

Comment (DimCitus) [reply docs/ref/reference.rst]: Hi @Bogdanp, you're right, I've updated the PR (rebased on top of current master) and fixed the reference. I'm also going to add a message about the current state of this PR. Thanks!

**Comment by DimCitus** [comment]
This PR leaves several problems open, so while it's interesting as a prototype and opens the discussion, it's not currently considered as a good solution for the problem at hand:

  - exposing a public API in the command line that might be best though as a debugging facility, namely the commands `pg_autoctl do show ipaddr|cidr` would make more sense, maybe, than having them exposed as in this PR.
  - it might be better to use `samenet` in pg_hba.conf, see https://www.postgresql.org/docs/current/auth-pg-hba-conf.html ; and then we don't need to put down the hostname of the other host in the HBA file, and we bypass reverse DNS lookups entirely.
  - discovering which LAN IP address to use on a host with several IP addresses is going to be guess work at best, in the simple case where there's a single internal (non-routable) IP address that might be ok though, but the method we implemented in `fetchLocalIPAddress` (https://github.com/citusdata/pg_auto_failover/blob/master/src/bin/pg_autoctl/ipaddr.c#L39) feels like it might be too magic for its own good.
  - about `samenet`, some hosting facilities only provide internet routable IP addresses, we might want to refrain from granting connection to them in that case (but we could use a /32 as we do now when installing the IP address from the secondary node).

We might want to revisit the whole HBA management in pg_auto_failover and come up with a nicer overall implementation here, including facilities for users to provide their own HBA rules and then have pg_auto_failover only check that everything works as intended.
